src/main/java/uk/gov/ons/census/casesvc/utility/MsgDateHelper.java

```diff
@@ -0,0 +1,19 @@
+package uk.gov.ons.census.casesvc.utility;
+
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import org.springframework.messaging.Message;
+
+public class MsgDateHelper {
+  public static OffsetDateTime getMsgTimeStamp(Message<?> msg) {
+
+    if (msg.getHeaders().getTimestamp() == null) {
+      throw new RuntimeException(
+          "Message Headers missing Timestamp, is this important enough to reject message on?");
```

Comment:
Looks like a work in progress exception message, we need to decide if it is important enough to reject the message.